examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/conversations/Animal.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.conversation.conversations
+
+import com.xebia.functional.xef.conversation.llm.openai.OpenAI
+import com.xebia.functional.xef.conversation.llm.openai.prompt
+import com.xebia.functional.xef.prompt.Prompt
+import com.xebia.functional.xef.prompt.templates.system
+import com.xebia.functional.xef.prompt.templates.user
+import kotlinx.serialization.Serializable
+
+@Serializable data class Animal(val name: String, val habitat: String, val diet: String)
+
+@Serializable
+data class Invention(val name: String, val inventor: String, val year: Int, val purpose: String)
+
+suspend fun main() {
```

Comment:
I have changed this example. In the previous version, we directly added information about the animal and invention to the prompt. Now, this example uses this information in the earlier message of the conversation, showing that OpenAi is receiving all the messages in the conversation